src/XliffTasks.Tests/StringExtensionsTests.cs

```diff
@@ -0,0 +1,73 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Xunit;
+
+namespace XliffTasks.Tests
+{
+    public class StringExtensionsTests
+    {
+        [Fact]
+        public void GetReplacementCount_NoPlaceholders()
+        {
+            var text = "Alpha";
+            var replacementCount = text.GetReplacementCount();
+
+            Assert.Equal(expected: 0, actual: replacementCount);
+        }
+
+        [Fact]
+        public void GetReplacementCount_OneSimplePlaceholder()
+        {
+            var text = "{0}";
+            var replacementCount = text.GetReplacementCount();
+
+            Assert.Equal(expected: 1, actual: replacementCount);
+        }
+
+        [Fact]
+        public void GetReplacementCount_PlaceholderWithAlignment()
+        {
+            var text = "{0,-3}";
+            var replacementCount = text.GetReplacementCount();
+
+            Assert.Equal(expected: 1, actual: replacementCount);
+        }
+
+        [Fact]
+        public void GetReplacementCount_PlaceholderWithFormatString()
+        {
+            var text = "{0:N}";
+            var replacementCount = text.GetReplacementCount();
+
+            Assert.Equal(expected: 1, actual: replacementCount);
+        }
+
+        [Fact]
+        public void GetReplacementCount_WithAlignmentAndFormatString()
+        {
+            var text = "{0,-10:G1}";
+            var replacementCount = text.GetReplacementCount();
+
+            Assert.Equal(expected: 1, actual: replacementCount);
+        }
+
+        [Fact]
+        public void GetReplacementCount_MultiplePlaceholders()
+        {
+            var text = "{0} {1}";
+            var replacementCount = text.GetReplacementCount();
+
+            Assert.Equal(expected: 2, actual: replacementCount);
+        }
+
+        [Fact]
+        public void GetReplacementCount_MultipleIdenticalPlaceholders()
+        {
+            var text = "{2} {2}";
+            var replacementCount = text.GetReplacementCount();
+
+            Assert.Equal(expected: 3, actual: replacementCount);
```

Comment:
Why do we expect 3 here?

Reply:
What we really need to count is the number of items that need to get passed to `string.Format`, not the number of placeholders in the original text. A `{2}` placeholder implies that you need to pass at least three items to `string.Format`.

Reply:
oh i see.

Reply:
So {0} {1} {2} to be translated to {2} {2} {2} would pass? I think in terms of comparing translation to source, it would be fine to see that we have the same count..